hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -844,6 +847,43 @@ public MerkleUniqueToken getUniqueToken(final NftID id) {
 		return uniqueTokenSupplier.get().get(fromNftID(id));
 	}
 
+	@Override
+	public List<TokenNftInfo> getUniqueTokenInfosFor(AccountID id, long start, long end) {
+		List<TokenNftInfo> nfts = new ArrayList<>();
+		// TODO: Discuss index types, protos are int64, compiled to long, but invertibleMap works with int.
+		uniqueTokenSupplier.get().inverseGet(ownershipMock(id), (int)start, (int)end).forEachRemaining(u -> {
+			var nft = uniqueTokenSupplier.get().get(u);
+			var creationTime = nft.getCreationTime();
+			nfts.add(
+					TokenNftInfo.newBuilder()
+							.setAccountID(id)
+							.setCreationTime(Timestamp.newBuilder()
+									.setSeconds(creationTime.getSeconds())
+									.setNanos(creationTime.getNanos())
+									.build())
+							.setNftID(NftID.newBuilder()
+									.setTokenID(fromEntityId(u.tokenId()))
+									.setSerialNumber(u.serialNumber())
+									.build())
+							.setMetadata(ByteString.copyFrom(nft.getMetadata()))
+							.build()
+			);
+		});
+		return nfts;
+	}
+
+	private TokenID fromEntityId(EntityId e) {
+		return TokenID.newBuilder()
+				.setRealmNum(e.realm())
+				.setShardNum(e.shard())
+				.setTokenNum(e.num())
+				.build();
+	}
+
+	private MerkleUniqueToken ownershipMock(AccountID id) {
```

Comment:
I think that this method is unnecessary and `mocks` are usually something that you refer to in testing